### DISCUSSION TITLE
Boilerplate setup

### DISCUSSION THREAD

**Discussion by owendwyer** [pixijs/pixijs#7694] [General]
Hi guys,

I'm looking to move a bunch of projects from CreateJS to pixijs and I've been working on a base of boilerplate code to reuse.

However, I'm not really sure what I'm doing and think I may be making some big mistakes.

If anyone has the time, I'd really appreciate if you could take a look over and see if there is anything glaringly wrong.

https://github.com/owendwyer/pixi-ts-boilerplate

In particular, I'm concerned about the way that alternative sprites are used for different screen sizes - this is something that I've never bothered with before.  The code here detects screen size and loads a different sprite and sprite atlas accordingly.  The default size is 800x550 and then there are sprites at x1.3 and x1.6 size.  As far as I can tell, it's enough to append x1.3 to the sprite png filename and then pixi knows to adjust the size.

Also, I'm not really sure about devicePixelResolution.  The code here just detects this and passes the value to the renderer.  Although, the value is maxed to 2 because I read somewhere else that a high resolution value can cause performance issues.

**Comment by ivanpopelyshev** [comment]
> In particular, I'm concerned about the way that alternative sprites are used for different screen sizes - this is something that I've never bothered with before.

So far, no one made solution that is worthy pixijs, thats why we dont have anything embedded in lib.

My approach: 
1. if devices native (*devicePixelRatio) resolution is close by height to your target - just use it! Add paddings or extra background to scene if needed.
2. otherwise, choose 1x or 2x, and scale the canvas itself. Like, you want 600x400 but phone has only 400x300? make it 400x300 but with adjusted pixel ratio (fit)

Anyway, its your job to do those resizes, and if possible report the results back, maybe you'll be the hero who finally founds acceptable solution for everyone :)

**Comment by miltoncandelero** [comment]
> If anyone has the time, I'd really appreciate if you could take a look over and see if there is anything glaringly wrong.

Oh boi
In your package.json
- `"@types/pixi.js": "^4.8.8",` This is wrong. 4.x is old and you are using 5.x that includes types already. Remove this.
- `"gsap": "^3.6.0",` I personally hate the [gsap licence ](https://greensock.com/standard-license/), so much that I made [my own tween library](https://miltoncandelero.github.io/tweedle.js/)
- `"pixi.js": "^5.3.7",` You are starting from scratch, why not go for the latest `6.1.1`?
  - Overall, why don't you update all your libs? If you are starting from scratch, try starting from the latest version of everything. Run `npx npm-check -u` and update everything 🤔 
- `"eslint-plugin-jsx-a11y": "^6.3.1",` Are you going to use React in any measure? Anything referencing `jsx` or `tsx` means React.
- `"@types/howler": "^2.2.1",` You have the types for howler but you don't have the actual lib. Also, why not [use pixi sound](https://pixijs.io/sound/examples/)?

**Comment by owendwyer** [comment]
Thanks a lot guys.  I'm going to start using this slowly and see how it goes.

> Overall, why don't you update all your libs? If you are starting from scratch, try starting from the latest version of everything. Run npx npm-check -u and update everything thinking

Honestly, I just never really thought about the package.json much.  Your feedback is appreciated.
